src/components/MAPI/apps/utils.ts

```diff
@@ -847,15 +847,26 @@ export function formatYAMLError(err: unknown): string {
 
 /**
  * Determines whether an app has a newer version
- * @param app
- * @param client
+ * @param clientFactory
  * @param auth
+ * @param cache
+ * @param app
  */
 export async function hasNewerVersion(
-  app: applicationv1alpha1.IApp,
-  client: IHttpClient,
-  auth: IOAuth2Provider
+  clientFactory: HttpClientFactory,
+  auth: IOAuth2Provider,
+  cache: Cache,
+  app: applicationv1alpha1.IApp
 ): Promise<{ name: string; hasNewerVersion: boolean }> {
+  const catalogNamespace = await getCatalogNamespace(
+    clientFactory,
+    auth,
```

Comment:
Refactored the function for determining whether there is a newer version for an `app`: we also get the `catalog`'s namespace so we can list `catalogs` within a namespace.